src/hb-set.h

```diff
@@ -60,126 +60,147 @@ typedef struct hb_set_t hb_set_t;
 HB_EXTERN hb_set_t *
 hb_set_create (void);
 
+/**
+ * Creates a set initially populated with num_codepoint codepoints, which must
+ * be in increasing order.
+ *
+ * Since: REPLACEME
+ */
+HB_EXTERN hb_set_t *
+hb_set_create_from_sorted (const hb_codepoint_t  *sorted_codepoints,
+                           unsigned int          num_codepoints);
+
+/**
+ * Exports the contents of the set to an array of hb_codepoint_t.
+ * codepoints[0] .. codepoints[hb_set_get_population(set) - 1] will be
+ * written to; you must ensure the size of the memory codepoints points to.
+ * The results will be in ascending order.
+ *
+ * Since: REPLACEME
+ */
+HB_EXTERN void
+hb_set_export_array (const hb_set_t  *set,
+                     hb_codepoint_t  *codepoints);
```

Comment:
For safety the caller should tell us how big the array is so we don't overrun the buffer. You can have this just fill the buffer up to the maximum size if the provided buffer is undersized, maybe with a return code to indicate it was too small?

Reply:
Done